File: exercism/nim/bob/bob.nim
import std/[strutils]

func isAllCapital(message: string): bool =
  var hasCapital = false
  for c in message:
    if c.isAlphaAscii:
      if c notin {'A' .. 'Z'}:
        return false
      else:
        hasCapital = true
  return hasCapital

func hey*(message: string): string =
  let message = message.strip
  if message.len == 0:
    "Fine. Be that way!"
  elif message[message.len - 1] == '?':
    if message.isAllCapital:
      "Calm down, I know what I'm doing!"
    else:
      "Sure."
  elif message.isAllCapital:
    "Whoa, chill out!"
  else:
    "Whatever."

File: exercism/python/024-bob/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()

    if len(hey_bob) == 0:
        return "Fine. Be that way!"

    is_question = hey_bob[-1] == '?'
    is_yelling = hey_bob.isupper()

    if is_question and not is_yelling:
        return "Sure."
    elif is_question and is_yelling:
        return "Calm down, I know what I'm doing!"
    elif not is_question and is_yelling:
        return "Whoa, chill out!"
    else:
        return "Whatever."
